integrations/gcp/gcp_core/search/resource_searches.py

```diff
@@ -38,144 +45,157 @@ async def search_all_resources(
+        Search for resources that the caller has ``cloudasset.assets.searchAllResources`` permission on within the project's scope.
+        """
+        project_name = project["name"]
+        logger.info(f"Searching all {asset_type}'s in project {project_name}")
+
         search_all_resources_request = {
             "scope": project_name,
             "asset_types": [asset_type],
             "read_mask": "*",
         }
         if asset_name:
             search_all_resources_request["query"] = f"name={asset_name}"
-        try:
-            paginated_responses: pagers.SearchAllResourcesAsyncPager = (
-                await async_assets_client.search_all_resources(
-                    search_all_resources_request, retry=retry_policy
+
+        asset_rate_limiter = await search_all_resources_qpm_per_project[
+            project_name.split("/")[-1]
+        ]
+
+        async with AssetServiceAsyncClient() as async_assets_client:
+
+            try:
+                async for assets in paginated_query(
+                    async_assets_client,
+                    "search_all_resources",
+                    search_all_resources_request,
+                    parse_asset_response,
+                    rate_limiter=asset_rate_limiter,
+                ):
+                    yield assets
+
+            except PermissionDenied:
+                logger.error(
+                    f"Service account doesn't have permissions to search all resources within project {project_name} for kind {asset_type}"
+                )
+            except NotFound:
+                logger.debug(
+                    f"Couldn't perform search_all_resources on project {project_name} since it's deleted."
+                )
+            else:
+                logger.info(
+                    f"Successfully searched all resources within project {project_name}"
                 )
-            )
-            async for paginated_response in paginated_responses.pages:
-                raw_assets = parse_protobuf_messages(paginated_response.results)
-                assets = typing.cast(list[AssetData], raw_assets)
-                if assets:
-                    logger.info(f"Found {len(assets)} {asset_type}'s")
-                    latest_resources = []
-                    for asset in assets:
-                        latest_resource = parse_latest_resource_from_asset(asset)
-                        latest_resource[EXTRA_PROJECT_FIELD] = project
-                        latest_resources.append(latest_resource)
-                    yield latest_resources
-        except PermissionDenied as e:
-            logger.exception(
-                f"Couldn't access the API Cloud Assets to get kind {asset_type}. Please set cloudasset.assets.searchAllResources permissions for project {project_name}"
-            )
-            raise e
 
 
 async def list_all_topics_per_project(
-    project: dict[str, Any],
+    project: dict[str, Any], **kwargs: Any
```

Comment:
```suggestion
    project: dict[str, Any]
```

Reply:
yeah, that.. actually, its meant to be there, I'm passing the kind from here
<img width="1154" alt="Screenshot 2024-08-27 at 10 41 35 AM" src="https://github.com/user-attachments/assets/eee90921-915b-4edd-b317-3b0c13596b7e">